test/functional/test_cluster_basic.py

```diff
@@ -104,6 +105,27 @@ def test_slave_reconnection_does_not_take_down_master(self):
         self._assert_build_completed_as_expected(build_1['build_id'], test_config, project_dir)
         self._assert_build_completed_as_expected(build_2['build_id'], test_config, project_dir)
 
+    def test_failed_setup_does_not_kill_slave(self):
+        master = self.cluster.start_master()
+        slave = self.cluster.start_slave()
+
+        project_dir = tempfile.TemporaryDirectory()
+        build_resp = master.post_new_build({
+            'type': 'directory',
+            'config': yaml.safe_load(FAILING_SETUP_JOB.config[os.name])['FailingSetupJob'],
+            'project_directory': project_dir.name,
+        })
+
```

Comment:
Will this test still pass if atomization fails? Should we also assert that num_atoms on the build is greater than 0?